tests/handlers/archive/test_tar.py

```diff
@@ -86,24 +184,43 @@ def test_offset(contents: bytes, expected_length: int, message: str):
 
 
 @pytest.mark.parametrize(
-    "contents, expected_length, message",
+    "contents",
+    (
+        pytest.param(
+            GNU_TAR_CONTENTS,
+            id="gnu-tar",
+        ),
+        pytest.param(
+            POSIX_TAR_CONTENTS,
+            id="posix-tar",
+        ),
+    ),
+)
+@pytest.mark.parametrize(
+    "start_complete, message",
     (
         pytest.param(
-            TRUNCATED_TAR_CONTENTS,
-            -1,
+            False,
             "File is truncated and no content can be recovered",
             id="empty-truncated",
         ),
         pytest.param(
-            TAR_CONTENTS + TRUNCATED_TAR_CONTENTS,
-            len(TAR_CONTENTS),
+            True,
             "File is truncated but valid parts should be recovered",
             id="truncated",
         ),
     ),
 )
-def test_truncated_files(contents: bytes, expected_length: int, message: str):
-    f = io.BytesIO(contents)
+def test_truncated_files(contents: bytes, start_complete: bool, message: str):
```

Comment:
These are 2 completely different test cases. It would easier to follow and debug in case of a failure and the implementation would be 4x simpler if you would just implement them in 2 separate functions without any parametrization.

Before the change, only the `contents` was different, which is good for a parametrized test, but after the change literally everything changed, so it doesn't fit into one test case anymore.